protocols/kad/src/lib.rs

```diff
@@ -19,6 +19,23 @@
 // DEALINGS IN THE SOFTWARE.
 
 //! Implementation of the libp2p-specific Kademlia protocol.
+//!
+//! # Overview
+//! Kademlia provides a distributed hash table in which data can be stored. Kademlia can also be
+//! used to store information about peers in a p2p network.
```

Comment:
```suggestion
//!
//! See [specification](https://github.com/libp2p/specs/blob/master/kad-dht/README.md) for details.
```

I would rather just link to the specification, serving as a single source of truth.